app/controllers/HomeController.scala

```diff
@@ -66,19 +69,40 @@ abstract class HomeController(
 
   private lazy val maxAnalysisExerciseFileSizeInBytes = 4096 * 1024
 
+  // scalastyle:off cyclomatic.complexity
   def present(implicit displaySdipEligibilityInfo: Boolean = false): Action[AnyContent] = CSRSecureAction(ActiveUserRole) {
     implicit request =>
       implicit cachedData =>
-        cachedData.application.map { implicit application =>
-          cachedData match {
-            case _ if isPhase1TestsPassed && (isEdip(cachedData) || isSdip(cachedData)) => displayEdipOrSdipResultsPage
+        val process = for {
+        updatedCachedData <- env.userService.refreshCachedUser(cachedData.user.userID)(hc, request)
```

Comment:
I think this would be better placed in SecureActions so that the "cache" is always refreshed as we have this problem with stale data elsewhere.  Also, having it here means we always have two db calls when we only really want one.